core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -397,6 +401,30 @@ private[spark] class Executor(
     }
 
     override def run(): Unit = {
+      val currentUser =
+        taskDescription.properties.getProperty(SparkContext.SPARK_JOB_PROXY_USER, null)
+      val tokens =
+        taskDescription.properties.getProperty(SparkContext.SPARK_JOB_PROXY_TOKEN, null)
+
+      if (currentUser != null && currentUser.nonEmpty && tokens != null && tokens.nonEmpty) {
+        val ugi = UserGroupInformation.createRemoteUser(currentUser)
+        val creds = new Credentials()
+        tokens.split(SparkContext.SPARK_JOB_PROXY_TOKEN_DELIMITER)
+          .foreach(tokenStr => {
+            val delegationToken = new Token()
+            delegationToken.decodeFromUrlString(tokenStr)
+            creds.addToken(new Text(delegationToken.getService), delegationToken)
+          })
+        ugi.addCredentials(creds)
+        ugi.doAs[Unit](new PrivilegedAction[Unit] {
+          override def run(): Unit = runWithProxy()
+        })
```

Comment:
Decode token to Credential from TaskDescription and  add Credential to UGI then proxy job owner user.